File: Napakalakirb/lib/pruebaNapakalaki.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "prize.rb"
require_relative "monster.rb"
require_relative "bad_consequence.rb"
require_relative "treasure_kind.rb"

class PruebaNapakalaki
  
 @@monsters = Array.new
 @@MAX_VALUE = 5
 
 def self.NvMayor10
   resultado = Array.new
   @@monsters.each do |m|
     if(m.combatLevel>10)
       resultado << m
     end
   end
   return resultado
 end
 
 def self.BcLostLevel
   resultado = Array.new
   @@monsters.each do |m|
     bc = m.malrollo
     if(bc.levels != 0 && bc.nVisibleTreasures == 0 && bc.nHiddenTreasures == 0 )
       resultado << m
     end
   end
   return resultado
 end
 
   
   def self.GananciaMayor1Nivel
  resultado = Array.new
 
  @@monsters.each do |m|
    if (m.getLevelsGained > 1)
        resultado << m
    end
  end
  return resultado
 
end

def self.PierdesTesoro(tesoro)
   resultado = Array.new
   
   @@monsters.each do |m| 
     bc = m.malrollo
     if ( bc.specificVisibleTreasures.include?(tesoro) || bc.specificHiddenTreasures.include?(tesoro) )
       resultado << m
     end
   end
   return resultado
end
 
 def self.monstruos
   p1 = Prize.new(2,1)
   bc1 = BadConsequence.newLevelSpecificTreasures("Pierdes tu armaduta visible y otra oculta", 0, [TreasureKind::ARMOR], [TreasureKind::ARMOR])
   @@monsters << Monster.new("Byakhees de bonanza", 8, bc1, p1)
   
   p2 = Prize.new(1,1)
   bc2 = BadConsequence.newLevelSpecificTreasures("Embobados con el lindo primigenio te descartas de tu casco visible", 0, [TreasureKind::HELMET], Array.new )
   @@monsters << Monster.new("Tenochtitlan", 2, bc2, p2)
   
   p3 = Prize.new(1, 1)
   bc3 = BadConsequence.newLevelSpecificTreasures("El primordial bostezo contagioso. Pierdes el calzado visible", 0, [TreasureKind::SHOES], Array.new)
   @@monsters << Monster.new("El sopor de Dunwich", 2, bc3, p3)
   
   p4 = Prize.new(4, 1)
   bc4 = BadConsequence.newLevelSpecificTreasures("Te atrapa para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta 1 mano visible y 1 mano oculta", 0, [TreasureKind::ONEHAND], [TreasureKind::ONEHAND])
   @@monsters << Monster.new("Demonios de Magaluf", 2, bc4, p4)
   
   p5 = Prize.new(3, 1)
   bc5 = BadConsequence.newLevelNumberOfTreasures("Pierdes todos tus tesoros visibles", 0, @@MAX_VALUE, @@MAX_VALUE)
   @@monsters << Monster.new("El gorrón en el umbral", 13, bc5, p5)
   
   p6 = Prize.new(2, 1)
   bc6 = BadConsequence.newLevelSpecificTreasures("Pierdes la armadura visible", 0, [TreasureKind::ARMOR], Array.new)
   @@monsters << Monster.new("H.P. Munchcraft", 6, bc6, p6)
   
   p7 = Prize.new(1, 1)
   bc7 = BadConsequence.newLevelSpecificTreasures("Sientes bichos bajo la ropa.Descarta la armadura visible", 0, [TreasureKind::ARMOR], Array.new)
   @@monsters << Monster.new("Necrófago", 13, bc7, p7)
   
   p8 = Prize.new(3, 2)
   bc8 = BadConsequence.newLevelNumberOfTreasures("Pierdes 5 niveles y 3 tesoros visibles", 5, 3, 0)
   @@monsters << Monster.new("El rey de rosado", 11, bc8, p8)
   
   p9 = Prize.new(1, 1)
   bc9 = BadConsequence.newLevelNumberOfTreasures("Toses los pulmones y pierdes 2 niveles", 2, 0, 0)
   @@monsters << Monster.new("Flecher", 2, bc9, p9)
   
   p10 = Prize.new(2,1)
   bc10 = BadConsequence.newDeath("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto")
   @@monsters << Monster.new("Los hondos", 8, bc10, p10)

   p11 = Prize.new(2,1)
   bc11 = BadConsequence.newLevelNumberOfTreasures("Pierdes 2 niveles y 2 tesoros ocultos", 2, 0, 2)
   @@monsters << Monster.new("Semillas Cthulhu", 4, bc11, p11)

   p12 = Prize.new(2, 1)
   bc12 = BadConsequence.newLevelSpecificTreasures("Te intentas escaquear. Pierdes una mano visible", 0, [TreasureKind::ONEHAND] , Array.new)
   @@monsters << Monster.new("Dameargo", 1, bc12, p12)

   p13 = Prize.new(2, 1)
   bc13 = BadConsequence.newLevelNumberOfTreasures("Da mucho asquito. Pierdes 3 niveles", 3, 0, 0)
   @@monsters << Monster.new("Pollipolipo volante", 3, bc13, p13)

   p14 = Prize.new(3, 1)
   bc14 = BadConsequence.newDeath("No le hace gracia que pronuncien mal su nombre. Estas muerto")
   @@monsters << Monster.new("Yskhtihyssg-Goth", 14, bc14, p14)

   p15 = Prize.new(3, 1)
   bc15 = BadConsequence.newDeath("La familia te atrapa. Estas muerto")
   @@monsters << Monster.new("Familia feliz", 1, bc15, p15)

   p16 = Prize.new(2, 1)
   bc16 = BadConsequence.newLevelSpecificTreasures("La quinta directiva primaria te obliga a perder dos niveles y un tesoro 2 manos visible", 2, [TreasureKind::BOTHHANDS] , Array.new)
   @@monsters << Monster.new("Roboggoth", 8, bc16, p16)

   p17 = Prize.new(1, 1)
   bc17 = BadConsequence.newLevelSpecificTreasures("Te asustas en la noche. Pierdes un casco visible", 0, [TreasureKind::HELMET], Array.new)
   @@monsters << Monster.new("El espia sordo", 5, bc17, p17)

   p18 = Prize.new(2, 1)
   bc18 = BadConsequence.newLevelNumberOfTreasures("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles", 2, 5, 0)
   @@monsters << Monster.new("Tongue", 19, bc17, p17)
   
   p19 = Prize.new(2, 1)
   bc19 = BadConsequence.newLevelSpecificTreasures("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos", 3, [TreasureKind::ONEHAND , TreasureKind::BOTHHANDS] , Array.new)
   @@monsters << Monster.new("Bicéfalo", 21, bc19, p19)
   
 end
 
 def self.prueba
   
    p = Prize.new(5,9)
    bc = BadConsequence.newLevelSpecificTreasures("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos", 3, [TreasureKind::ONEHAND , TreasureKind::BOTHHANDS] , Array.new)
    m1 = Monster.new("Joseyo",8,bc,p)
 
   
    tesoro = "TreasureKind::ONEHAND"
    if(m1.malrollo.specificVisibleTreasures.include?(tesoro))
    puts "hola"
   end
 end
 
 PruebaNapakalaki.monstruos
 #puts PruebaNapakalaki.NvMayor10
 #puts PruebaNapakalaki.BcLostLevel
 #puts PruebaNapakalaki.GananciaMayor1Nivel
puts PruebaNapakalaki.PierdesTesoro(TreasureKind::ONEHAND)
# PruebaNapakalaki.prueba
end
 

File: Napakalaki/src/napakalaki/PruebaNapakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;
import java.util.Arrays;

/**
 *
 * @author jose
 */
public class PruebaNapakalaki {
    
    static ArrayList<Monster> monstruos = new ArrayList();
    
    static ArrayList<Monster> NvMayor10(){
        ArrayList<Monster> salida = new ArrayList();
        for( Monster m : monstruos){
            if(m.getCombatLevel()>10){
                salida.add(m);
            }
        }
        return salida;     
    }
    
    static ArrayList<Monster> BcLostLevel(){
        ArrayList<Monster> salida = new ArrayList();
        for( Monster m : monstruos){
            BadConsequence p = m.getBadConsequence();
            if(p.getLevels()>0 ){
                salida.add(m);
            }
        }
        return salida;
    }
    
    static public ArrayList<Monster>GananciaMayor1Nivel(){
            ArrayList<Monster>salida = new ArrayList();
            for( Monster m: monstruos ){
                if( (m.getLevelsGained()) > 1 )
                    salida.add(m);
                   
            }
    return salida;
    }
    
    /*static public ArrayList<Monster>PierdesTesoro(TreasureKind Tesoro){
            ArrayList<Monster>salida = new ArrayList();
            for( Monster m: monstruos ){
                if( (m.getBadConsequence().getspecificHiddenTreasures().contains(Tesoro)) || m.getBadConsequence().getspecificVisibleTreasures().contains(Tesoro))
                     salida.add(m);
            }
            return salida;
    }*/
    
    
    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        SpecificBadConsequence bc1 = new SpecificBadConsequence("Pierdes tu armadura visible y otra oculta.", 0, new ArrayList(Arrays.asList(TreasureKind.ARMOR)),new ArrayList(Arrays.asList(TreasureKind.ARMOR)) );
        Prize prize1 = new Prize(2,1);
        monstruos.add(new Monster("3 Byakhees de bonanza", 8, bc1,prize1));

        SpecificBadConsequence bc2 = new SpecificBadConsequence("Embobados con el lindo primigenio te descartas tu casco visible",0,new ArrayList(Arrays.asList(TreasureKind.HELMET)), new ArrayList());
        Prize prize2 = new Prize(1,1);
        monstruos.add(new Monster("Tenochtitlan",2,bc2,prize2));

        SpecificBadConsequence bc3 = new SpecificBadConsequence("El primordial bostezo contagioso. Pierdes el calzado visible.",0,new ArrayList(Arrays.asList(TreasureKind.SHOES)),new ArrayList());
        Prize prize3 = new Prize(1,1);
        monstruos.add(new Monster("El sopor de Dunwich",2,bc3,prize3));

        SpecificBadConsequence bc4 = new SpecificBadConsequence("Te atrapan para llevarte de fiesta y te dejan caer en mitad del vuelo. Descarta a 1 mano visible y a 1 mano oculta.",0,new ArrayList(Arrays.asList(TreasureKind.ONEHAND)),new ArrayList(Arrays.asList(TreasureKind.ONEHAND)));
        Prize prize4 = new Prize(4,1);
        monstruos.add(new Monster("Demonios de Magalud",2,bc4,prize4));

        NumericBadConsequence bc5 = new NumericBadConsequence("Pierdes todos tus tesoreos visibles.",0,BadConsequence.getMaxTreasures(),0);
        Prize prize5 = new Prize(3,1);
        monstruos.add(new Monster("El gorrón en el umbral",13,bc5,prize5));

        SpecificBadConsequence bc6 = new SpecificBadConsequence("Pierdes la armadura visible",0,new ArrayList(Arrays.asList(TreasureKind.ARMOR)),new ArrayList());
        Prize prize6 = new Prize(2,1);
        monstruos.add(new Monster("H.P. Munchcraft",6,bc6,prize6));

        SpecificBadConsequence bc7 = new SpecificBadConsequence("Sientes bichos bajo la ropa. Descarta la armadura visible.",0,new ArrayList(Arrays.asList(TreasureKind.ARMOR)),new ArrayList());
        Prize prize7 = new Prize(1,1);
        monstruos.add(new Monster("Necrófago",13,bc7,prize7));

        NumericBadConsequence bc8 = new NumericBadConsequence("Pierdes 5 niveles y 3 tesoros visibles.",5,3,0);
        Prize prize8 = new Prize(3,2);
        monstruos.add(new Monster("El rey de rosado",11,bc8,prize8));

        NumericBadConsequence bc9 = new NumericBadConsequence("Toses los pulmones y pierdes 2 niveles.",2,0,0);
        Prize prize9 = new Prize(1,1);
        monstruos.add(new Monster("Flecher",2,bc9,prize9));

        DeathBadConsequence bc10 = new DeathBadConsequence("Estos monstruos resultan bastante superficiales y te aburren mortalmente. Estas muerto",true);
        Prize prize10 = new Prize(2,1);
        monstruos.add(new Monster("Los hondos",8,bc10,prize10));

        NumericBadConsequence bc11 = new NumericBadConsequence("Pierdes 2 niveles y 2 tesoros ocultos.",2,0,2);
        Prize prize11 = new Prize(2,1);
        monstruos.add(new Monster("Semillas Cthulhu",4,bc11,prize11));

        SpecificBadConsequence bc12 = new SpecificBadConsequence("Te intentas escaquear. Pierdes una mano visible.",0,new ArrayList(Arrays.asList(TreasureKind.ONEHAND)),new ArrayList());
        Prize prize12 = new Prize(2,1);
        monstruos.add(new Monster("Dameargo",1,bc12,prize12));

        NumericBadConsequence bc13 = new NumericBadConsequence("Da mucho asquito. Pierdes 3 niveles.",3,0,0);
        Prize prize13 = new Prize(2,1);
        monstruos.add(new Monster("Pollipólipo volante",3,bc13,prize13));

        DeathBadConsequence bc14 = new DeathBadConsequence("No le hace gracia que pronuncien mal su nombre. Estas muerto", true);
        Prize prize14 = new Prize(3,1);
        monstruos.add(new Monster("Yskhtihyssg-Goth",14,bc14,prize14));

        DeathBadConsequence bc15 = new DeathBadConsequence("La familia te atrapa. Estás muerto",true);
        Prize prize15 = new Prize(3,1);
        monstruos.add(new Monster("Familia feliz",1,bc15,prize15));

        SpecificBadConsequence bc16 = new SpecificBadConsequence("La quinta directiva primaria te obliga a perder 2 niveles y un tesoro 2 manos visible",2,new ArrayList(Arrays.asList(TreasureKind.BOTHHANDS)),new ArrayList());
        Prize prize16 = new Prize(2,1);
        monstruos.add(new Monster("Roboggoth",8,bc16,prize16));

        SpecificBadConsequence bc17 = new SpecificBadConsequence("Te asusta la noche. Pierdes un casco visible",0,new ArrayList(Arrays.asList(TreasureKind.HELMET)),new ArrayList());
        Prize prize17 = new Prize(1,1);
        monstruos.add(new Monster("El espía sordo",5,bc17,prize17));

        NumericBadConsequence bc18 = new NumericBadConsequence("Menudo susto te llevas. Pierdes 2 niveles y 5 tesoros visibles",2,5,0);
        Prize prize18 = new Prize(2,1);
        monstruos.add(new Monster("Tongue",19,bc18,prize18));

        SpecificBadConsequence bc19 = new SpecificBadConsequence("Te faltan manos para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles de las manos",3,new ArrayList(Arrays.asList(TreasureKind.BOTHHANDS,TreasureKind.ONEHAND)),new ArrayList());
        Prize prize19 = new Prize(2,1);
        monstruos.add(new Monster("Bicéfalo",21,bc19,prize19));
        
        
        //System.out.println(PruebaNapakalaki.PierdesTesoro(TreasureKind.ONEHAND).toString());
        System.out.println(PruebaNapakalaki.GananciaMayor1Nivel().toString());
            
    }
    
}
